4.x - Add typing to Database/Statement

markstory opened the pull request:
I had to widen the types on fetch() and fetchAll() as we not only take the CakePHP constants, but also PDO constants which are integers.

Refs #11935

markstory committed:
Add typing to Database/Statement

I had to widen the types on fetch() and fetchAll() as we not only take
the CakePHP constants, but also PDO constants which are integers.

ADmad commented on src/Database/Statement/BufferedStatement.php:
```diff
@@ -337,7 +338,7 @@ public function next()
      * @return \Cake\Database\StatementInterface
      */
-    public function getInnerStatement()
+    public function getInnerStatement(): \Cake\Database\StatementInterface
```
Should add "use" statement instead of using FQCN. There are other such cases to in the PR.

dereuromark replied:
There are sniffers that can automate this. We have one in spryker namespace for example.

markstory replied:
I ran a tool, and didn't see this when I was cleaning up. I'll fix this now.

markstory committed:
Fix type inconsistencies.

We need to accept PDO constants in all these methods, so we can't use
string constants.

markstory committed:
Add missing type.
